src/Core/Framework/DependencyInjection/api.xml

```diff
@@ -197,8 +197,10 @@
             <argument type="service" id="cache.object"/>
             <argument type="service" id="Shopware\Storefront\Framework\Cache\CacheWarmer\CacheWarmer" on-invalid="null"/>
             <argument type="service" id="Shopware\Core\Framework\DataAbstractionLayer\Indexing\EntityIndexerRegistry"/>
+            <tag name="container.service_subscriber" />
+            <tag name="controller.service_arguments" />
             <call method="setContainer">
-                <argument type="service" id="service_container"/>
+                <argument type="service" id="Psr\Container\ContainerInterface" />
```

Comment:
not sure if I should do it this way (correct service definition) or temporarily override the `getParameter` method and check if the full service container is injected.

However, I doubt this will be a breaking change because one would need to override the controller and actually retrieve services from the service container but if you fear that might be the case, I will change it.